### PR TITLE
WIP: Why your probably shouldnt add a CLA to your open source project

### PR DESCRIPTION
**Pull request by benbalter**
Fixes https://github.com/benbalter/benbalter.github.com/issues/444.

cc @mikemcquaid

### CODE REVIEW

File: _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+Unlike licenses, however, contributor license agreements, or CLAs, are not standardized, meaning if you're a contributor, you'll have to read each CLA to determine what legal rights you're giving away, before contributing (and hope you can parse what's often dense legalese if you're not a lawyer).
+
+### Why projects adopt a CLA
+
+CLAs are generally a form of CYA, to prevent maintainers (or the project) from landing in hot water due to the community contributions it accepts. They are an extremely effective way to reduce a maintainer's exposure to intellectual property liability, especially if that maintainer is a for-profit business. Specifically, it makes ownership of the project's intellectual property explicit.
```
Comment (MikeMcQuaid) [inline _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md]: Worth defining `CYA`? Took me a minute to parse.

File: _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Think about it this way: the role of an attorney, especially a transactional attorney, is to minimize legal risk. For the vast majority of intellectual property transfers in the business world, that means lots of corporate paper and due diligence. When one company buys another company's software, be it a procurement or an acquisition, it often involves lengthy negotiations resulting in contracts with tens or hundreds of pages to ensure both parties exchange exactly what they intended to transfer, no more, no less.
+
+Contrast that with open source licensing, which often caries equal legal weight, but almost exclusively without the legal safeguards. With that "traditional" baseline in mind, it's natural then, for corporate counsels to suggest the company take steps to "paper" the transaction. But as mentioned earlier, legal's role is to advise business owners how to minimize risk, not how best to run an open source project, and such steps must be balanced against business motivations and the likelihood that the harm might occur.
+
+###
```
Comment (MikeMcQuaid) [inline _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md]: Missing a header?

File: _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
+title: Why your probably shouldn't add a CLA to your open source project
```
Comment (IanLee1521) [inline _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md]: "Why your..." -> "Why you..."  (minus the `r`)

File: _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+description:
+---
+
+Contributor license agreements or CLAs for short have gained a lot of visibility in recent years as many prominent open source projects have opted to use them. If all the cool kids are doing it, should you? Probably not. Here's why:
```
Comment (IanLee1521) [inline _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md]: Maybe include links to some of the prominent projects using CLAs ?

File: _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+
+### What is a CLA?
+
+First, if you're new to concepts like copyright, patents, and licensing, this post on [everything an open source maintainer might need to know about open source licensing](https://ben.balter.com/2017/11/28/everything-an-open-source-maintainer-might-need-to-know-about-open-source-licensing/) may provide some helpful background.
+
+In short, you can think of a CLA as somewhat of a "reverse open source license". Whereas open source licenses are a copyright grant for users to use the project, a contributor license agreement, at its core, is the right for the project to incorporate a contributor's code. Contributors may also be required to attest that they have the right to submit the code, or the CLA may include an explicit patent grant. Some CLAs go so far as to actually assign the developers copyright to the project.
+
+Unlike licenses, however, contributor license agreements, or CLAs, are not standardized, meaning if you're a contributor, you'll have to read each CLA to determine what legal rights you're giving away, before contributing (and hope you can parse what's often dense legalese if you're not a lawyer).
```
Comment (IanLee1521) [inline _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md]: Might also mention that things can get especially complicated if you're working for a company and contributing as an employee.. That can mean the company may need to read and agree to each CLA as well.

File: _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+Think about it this way: the role of an attorney, especially a transactional attorney, is to minimize legal risk. For the vast majority of intellectual property transfers in the business world, that means lots of corporate paper and due diligence. When one company buys another company's software, be it a procurement or an acquisition, it often involves lengthy negotiations resulting in contracts with tens or hundreds of pages to ensure both parties exchange exactly what they intended to transfer, no more, no less.
+
+Contrast that with open source licensing, which often caries equal legal weight, but almost exclusively without the legal safeguards. With that "traditional" baseline in mind, it's natural then, for corporate counsels to suggest the company take steps to "paper" the transaction. But as mentioned earlier, legal's role is to advise business owners how to minimize risk, not how best to run an open source project, and such steps must be balanced against business motivations and the likelihood that the harm might occur.
```
Comment (IanLee1521) [inline _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md]: typo: "caries" -> "carries"

File: _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+### 
+
+*  CLAs simply shift legal blame for any patent infringement, copyright infringement, or other bad acts from the project (or its legal entity) back onto its contributors.
+* CLA-assent records requires massive administrative overhead.
```
Comment (IanLee1521) [inline _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md]: typo: "requires" -> "require"

File: _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+* CLAs require the first interaction between a FLOSS project and a new contributor to involve a complex legal negotiation and a formal legal agreement.
```
Comment (IanLee1521) [inline _posts/2017-12-04-why-your-probably-shouldnt-add-a-cla-to-your-open-source-project.md]: First use of of "FLOSS", may want to define / sync with prior use of simply "open source projects"

**Comment by IanLee1521** [comment]
Love the idea! Definitely have some challenging discussions about CLAs here that we're still going through...

**Comment by benbalter** [comment]
@IanLee1521 @MikeMcQuaid Thanks for the 👀. Took longer than expected (and ultimately a slightly different direction), but glad to get this off my list! :smile:

**Comment by benbalter** [comment]
https://ben.balter.com/2018/01/02/why-you-probably-shouldnt-add-a-cla-to-your-open-source-project/

**Pull request merged**